clang/lib/Frontend/CompilerInvocation.cpp

```diff
@@ -3565,6 +3583,65 @@ void CompilerInvocation::GenerateLangArgs(const LangOptions &Opts,
     GenerateArg(Args, OPT_fno_gpu_rdc, SA);
 }
 
+void CompilerInvocation::ParseFpAccuracyArgs(LangOptions &Opts, ArgList &Args,
+                                             DiagnosticsEngine &Diags) {
+  for (StringRef Values : Args.getAllArgValues(OPT_ffp_accuracy_attr_EQ)) {
+    SmallVector<StringRef, 8> ValuesArr;
+    Values.split(ValuesArr, ' ');
+    for (const auto &Val : ValuesArr) {
+      SmallVector<StringRef, 3> ValElement;
+      Val.split(ValElement, ':');
+      // The option is of the form -ffp-accuracy=value.
+      if (ValElement.size() == 1) {
+        StringRef FPAccuracy = ValElement[0];
+        if (!(FPAccuracy.equals("default") || FPAccuracy.equals("high") ||
+              FPAccuracy.equals("low") || FPAccuracy.equals("medium") ||
+              FPAccuracy.equals("sycl") || FPAccuracy.equals("cuda")))
+          Diags.Report(diag::err_drv_unsupported_option_argument)
+              << "ffp-accuracy" << FPAccuracy;
+        std::pair<LangOptions::FPAccuracyMapTy::iterator, bool> Result =
+            Opts.FPAccuracyMap.insert({"fp-accuracy", FPAccuracy.str()});
```

Comment:
I don't understand what FPAccuracyMap is mapping. What is the difference between this and FPAccuracyFuncMap?

Reply:
FPAccuracyMap creates a map between fp-accuracy and the value of accuracy given of the command line.
For example: clang -c -ffp-accuracy=high t1.c
will create FPAccuracyMap with:
[fp-accuracy -> high]

FPAccuracyFuncMap will create a map between the functions on the command line and an FPAccuracyMap.
For example: clang -c -ffp-accuracy=high:[f1,f2] -ffp-accuracy=low:[f3] t1.c
will create a FPAccuracyFunMap with:
[f1 -> [fp-accuracy ->high]]
[f2 -> [fp-accuracy ->high]]
[f3 -> [fp-accuracy ->low]]

Reply:
So is there ever more than one entry in FPAccuracyMap?